src/aeso_syntax.erl

```diff
@@ -146,8 +145,7 @@
 -type stmt() :: letbind()
               | expr().
 
--type alt() :: {'case', ann(), pat(), expr()}
-             | {'case', ann(), pat(), expr(), expr()}.
+-type alt() :: {'case', ann(), pat(), [expr()], [expr()]}.
```

Comment:
why list of expr?

Reply:
multiple guards, each guard corresponds to one body.

Reply:
multiple guards, each guard corresponds to one body.

Reply:
I don't get it, how do you represent
```
switch(x)
  (a, b) | a > 0, b > 0 => a + b
         | true => 0 
```
?

Reply:
Or if it is invalid
```
switch(x)
  (a, b) | a > 0, b > 0 => a + b
  _  => 0 
```

Reply:
The above does is not valid since the guard should be a single expr, so if you write `a > 0 && b > 0` instead of `a > 0, b > 0` in the first example it would be parsed like this:

```
[{switch,                                                                                     [3/4542]
     [{file,no_file},{line,4},{col,5}],
     {id,[{file,no_file},{line,4},{col,12}],"x"},
     [{'case',
          [{file,no_file},{line,6},{col,26}],
          {tuple,
              [{file,no_file},{line,5},{col,7}],
              [{id,[{file,no_file},{line,5},{col,8}],"a"},
               {id,[{file,no_file},{line,5},{col,11}],"b"}]},
          [{app,
               [{file,no_file},{line,6},{col,11},{format,infix}],
               {'&&',[{file,no_file},{line,6},{col,17}]},
               [{app,
                    [{file,no_file},{line,6},{col,11},{format,infix}],
                    {'>',[{file,no_file},{line,6},{col,13}]},
                    [{id,[{file,no_file},{line,6},{col,11}],"a"},
                     {int,[{file,no_file},{line,6},{col,15}],0}]},
                {app,
                    [{file,no_file},{line,6},{col,20},{format,infix}],
                    {'>',[{file,no_file},{line,6},{col,22}]},
                    [{id,[{file,no_file},{line,6},{col,20}],"b"},
                     {int,[{file,no_file},{line,6},{col,24}],0}]}]},
           {bool,[{file,no_file},{line,7},{col,11}],true}],
          [{app,
               [{file,no_file},{line,6},{col,29},{format,infix}],
               {'+',[{file,no_file},{line,6},{col,31}]},
               [{id,[{file,no_file},{line,6},{col,29}],"a"},
                {id,[{file,no_file},{line,6},{col,33}],"b"}]},
           {int,[{file,no_file},{line,7},{col,19}],0}]}]}]}]}]
```

Notice that there are 2 guards and 2 bodies.

Do you think there's something wrong with that?

Reply:
Uh, when talking about multiple guards we were talking about `| a > 0, b > 0`, so in the future we might have `| a > 0, b > 0, let x = a + b, x > 10`. That's why there would be a list of guards.

Multiple guard alternatives for a single pattern are great too, but then why not pack them into a list of tuples? They are conceptually together, so imo they should stay side by side. 

My idea for the typing:

```erlang
-type alt() :: {'case', ann(), pat(), [guard_alt()]}.
-type guard_alt :: {guarded, ann(), [guard()], expr()}.
-type guard() :: expr().
```

This way this:
```
switch(x)
  (a, b) | a > 0, b > 0 => a + b
         | true => 0 
```
can be expressed as (simplified for readability)
```
{switch, _, "x", [
  {case, _, {tuple, ["a", "b"]}, [
    {guarded, _, [a > 0, b > 0], a + b},
    {guarded, _, true, 0}
  ]}
]}
```